### PR TITLE
Fix accessibility issue for instacart where tab or shift + tab move f…

### PR DESCRIPTION
**Pull request by ywang-branch**
JIRA: https://branch.atlassian.net/browse/BUX-930

Fix below accessibility issue for Instacart:
Screen Shot 2022-04-28 at 12 22 33 PM

### CODE REVIEW

File: src/journeys_utils.js [hunk 1]
```diff
@@ -266,12 +266,44 @@ journeys_utils.addHtmlToIframe = function(iframe, html, userAgent) {
 	else {
 		bodyClass = 'branch-banner-desktop';
 	}
-
+	var instacartAccessibilityTemplateId = '1042917133147779073';
```
Comment (csimpson-branch) [inline src/journeys_utils.js:269]: 1. Let's remove instacart's name from the variable.
2. We should also make it an array if we need to quickly add other templates.

Comment (csimpson-branch) [inline src/journeys_utils.js:269]:
```suggestion
	var accessibleTemplateIds = ['1042917133147779073'];
```

File: src/journeys_utils.js [hunk 2]
```diff
@@ -266,12 +266,44 @@ journeys_utils.addHtmlToIframe = function(iframe, html, userAgent) {
...
 	var iframedoc = iframe.contentDocument || iframe.contentWindow.document;
 	iframedoc.head = iframedoc.createElement('head');
 	iframedoc.body = iframedoc.createElement('body');
 	iframedoc.body.innerHTML = html;
 	iframedoc.body.className = bodyClass;
+	if (templateId === instacartAccessibilityTemplateId) {
```
Comment (csimpson-branch) [inline src/journeys_utils.js:275]:
```suggestion
	if (accessibleTemplateIds.includes(templateId) {
```

**Review by csimpson-branch — CHANGES_REQUESTED** [review]
Posted suggested changes. Thanks!

**Review by csimpson-branch — APPROVED** [review]
LGTM. It would be good to hear from SDK team member as well.

**Comment by ywang-branch** [comment]
Team, I met with Joerg and went over the changes. Because this is a public repo, he suggested to add a meta tag in the Instacart's html and in the web sdk repo we check for the content of the meta tag to restrict the script to run instead of exposing the instacart id. 

I think this is a great suggestion, what do you guys think?

**Comment by csimpson-branch** [comment]
@ywang-branch Did you validate the dashboard service doesn't sanitize `<meta>` tags?

**Review by csimpson-branch — CHANGES_REQUESTED** [review]
@ywang-branch Did you validate the dashboard service doesn't sanitize <meta> tags?

**Comment by ywang-branch** [comment]
> 



> @ywang-branch Did you validate the dashboard service doesn't sanitize tags?

I verified and tested the changes, there are no issues.

**Comment by jwangler-branch** [comment]
From my side, it looks good. 👍

**Review by csimpson-branch — APPROVED** [review]

File: src/journeys_utils.js [hunk 3]
```diff
@@ -255,7 +255,7 @@ journeys_utils.createIframe = function() {
  * @param {string} html - raw Journey HTML
  * @param {string} userAgent - UA to determine body class
  */
-journeys_utils.addHtmlToIframe = function(iframe, html, userAgent) {
+journeys_utils.addHtmlToIframe = function(iframe, html, userAgent, templateId) {
```
[the remark this answers is not in the document]
Comment (ywang-branch) [reply src/journeys_utils.js]: Good catch, I've updated the code. Please take a look again. Thanks

File: src/journeys_utils.js [hunk 4]
```diff
@@ -266,12 +266,44 @@ journeys_utils.addHtmlToIframe = function(iframe, html, userAgent) {
...
 	iframedoc.head = iframedoc.createElement('head');
 	iframedoc.body = iframedoc.createElement('body');
 	iframedoc.body.innerHTML = html;
 	iframedoc.body.className = bodyClass;
+	var metaTag = iframedoc.querySelector('meta[name="accessibility"]');
+	if (metaTag && metaTag.content === 'wcag') {
+		var scriptTag = iframedoc.createElement('script');
+		scriptTag.type = 'text/javascript';
+		scriptTag.text = `
+			const  focusableElements =
```
Comment (JagadeeshKaricherla-branch) [inline src/journeys_utils.js:279]: small/minor change : change const to var. Our closure compiler is very old and We have not yet tested ES6 features fully.So till the time we upgrade our closure compiler and fully test ES6 support, better to use var.

File: src/journeys_utils.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ywang-branch) [reply src/journeys_utils.js]: This is a very good point, I've updated the code. Please take a look again, thanks!

File: src/journeys_utils.js (hunk 4 above)
Comment (ywang-branch) [reply src/journeys_utils.js]: This is a very good point, I've updated the code. Please take a look again, thanks!

**Comment by JagadeeshKaricherla-branch** [comment]
@jwangler-branch : Few questions:

- Are changes to `api_endpoint, link_service_endpoint, app_service_endpoint ` in dist/build.js looking good ? I see instacart specific endpoints like api_endpoint changed from "https://api2.branch.io" to "https://instacart.api.beta.branch.io"

- Should we directly edit in dist/build.js or change them in https://github.com/BranchMetrics/web-branch-deep-linking-attribution/blob/492503c579624a4827a1cc18ac79751e85ab8730/src/0_config.js#L8 and run make file to generate build.js ?

File: dist/build.js [hunk 5]
```diff
@@ -958,7 +958,7 @@ goog.json.Serializer.prototype.serializeObject_ = function(a, b) {
   b.push("}");
 };
 // Input 2
-var config = {app_service_endpoint:"https://app.link", link_service_endpoint:"https://bnc.lt", api_endpoint:"https://api2.branch.io", version:"2.59.0"};
+var config = {app_service_endpoint:"https://app.stage.branch.io", link_service_endpoint:"https://link.stage.branch.io", api_endpoint:"https://instacart.api.beta.branch.io", version:"2.59.0"};
```
Comment (jwangler-branch) [inline dist/build.js:961]: Please create a new build by using make @ywang-branch 
we have to restore the original URL's

**Review by jwangler-branch — APPROVED** [review]

**Pull request merged**